The slf-atreus layout for the Keyboardio Atreus

PoweredByPorridge opened the pull request:
## slf-atreus layout

The slf (Stiff Little Fingers) layout helps me type with my slightly creaky fingers. It duplicates Control and Alt keys on the left and right, uses autoshift, and tap dances to minimize movement around the keyboard.

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None.

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

This pull request is the replacement for https://github.com/qmk/qmk_firmware/pull/20883 which was closed as invalid. 

Thanks, Duncan.

drashna commented on keyboards/keyboardio/atreus/keymaps/slf-atreus/config.h:
```diff
@@ -0,0 +1,55 @@
+
+#define TAPPING_TERM_PER_KEY
+
+// Prevent normal rollover on alphas from accidentally triggering mods.
+#define IGNORE_MOD_TAP_INTERRUPT
```
This is being deprecated (is so, on develop)

PoweredByPorridge replied:
I have removed this line and the comment. It was not used (no get_ignore_mod_tap_interrupt) in the code.

drashna commented on keyboards/keyboardio/atreus/keymaps/slf-atreus/config.h:
```diff
@@ -0,0 +1,55 @@
+// Prevent normal rollover on alphas from accidentally triggering mods.
+#define IGNORE_MOD_TAP_INTERRUPT
+
+// Enable rapid switch from tap to hold, disables double tap hold auto-repeat.
+#define TAPPING_FORCE_HOLD
```
Replaced by `QUICK_TAP_TERM`.

PoweredByPorridge replied:
I have removed this line and the comment. It was not used (no get_tapping_force_hold) in the code.

drashna commented on keyboards/keyboardio/atreus/keymaps/slf-atreus/keymap.c:
```diff
@@ -0,0 +1,266 @@
+    td_state = cur_dance(state);
+    switch (td_state) {
+        case SINGLE_TAP:
+            register_mods(MOD_BIT(KC_LSFT)); // For a layer-tap key, use `layer_on(_MY_LAYER)` here
+            register_code16(KC_9);
```
You can do this, instead: 
```suggestion
            register_code16(LSFT(KC_9));
```

PoweredByPorridge commented:
Hi @drashna. I have made the change that you requested in the line you pointed out, and in the other (5) places that I used the same code in keymap.c 

I removed some commented out stuff on home row mods that I experimented with a while ago. Lines 19-29.

I also removed the combos. So have made changes to config.h and rules.mk.

I don't know whether this falls within the scope of changes that should be made in a pull request; I suspect not. Should I cancel this pull request and start a new one? Or is that simply making more work for people like you who review them?

Thanks for your patience,

Duncan.

PoweredByPorridge commented:
Hi @drashna. I have made some minor updates to the comments in config.h. That appears to have unblocked/triggered the CI build which had been stuck since my last commit a few days ago.

Thanks, Duncan.

fauxpark commented on keyboards/keyboardio/atreus/keymaps/slf-atreus/rules.mk:
```diff
@@ -0,0 +1,30 @@
+# Bootloader selection
+# BOOTLOADER        = qmk-dfu  # Elite C
+BOOTLOADER        = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE  = no       # was = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE   = no       # Mouse keys
+EXTRAKEY_ENABLE   = yes      # Audio control and System control
+
+CONSOLE_ENABLE    = no       # Console for debug
+COMMAND_ENABLE    = no       # Commands for debug and configuration
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+
+SLEEP_LED_ENABLE  = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE       = yes      # USB Nkey Rollover
+BACKLIGHT_ENABLE  = no       # Enable keyboard backlight functionality
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE   = no       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE  = no       # Enable Bluetooth
+AUDIO_ENABLE      = no       # Audio output
+
+UNICODE_ENABLE    = no       # Unicode
+
+AUTO_SHIFT_ENABLE = yes      # Auto shift
+TAP_DANCE_ENABLE  = yes      # tap dance. One tap one code, two taps another code, three taps, and so on.
+
+# COMBO_ENABLE      = yes      # Combos - q+w=Esc
```
```suggestion
BOOTMAGIC_ENABLE = no
MOUSEKEY_ENABLE = no
CONSOLE_ENABLE = no
COMMAND_ENABLE = no
UNICODE_ENABLE = no
AUTO_SHIFT_ENABLE = yes
TAP_DANCE_ENABLE = yes
```

fauxpark reviewed (changes requested):
Keymap name should be your GitHub username, and may not contain `-` characters.

drashna commented:
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/chibios
git checkout qmk-master
```

PoweredByPorridge commented:
@drashna. My apologies, I didn't spot your comment, a combination of work and a non-functioning computer. I now appear to be back in the state I was a month ago, but am not entirely sure where I should go from here. 

Thanks, Duncan.

PoweredByPorridge commented:
I'm going to close this pull request and start again. I have made a mess and shall go back to the start. Thank you for your help.
